code/modules/pda/PDA.dm

```diff
@@ -295,18 +294,22 @@ GLOBAL_LIST_EMPTY(PDAs)
 		SStgui.update_uis(src)
 		playsound(src, 'sound/machines/pda_button1.ogg', 50, TRUE)
 	else if(is_pen(C))
-		var/obj/item/pen/O = locate() in src
-		if(O)
+		if(held_pen)
 			to_chat(user, "<span class='notice'>There is already a pen in \the [src].</span>")
 		else
 			user.drop_item()
 			C.forceMove(src)
+			held_pen = C
+			RegisterSignal(held_pen, COMSIG_PARENT_QDELETING, .proc/clear_pen)
```

Comment:
Prob best to make a `add_pen(obj/item/pen/P)` proc which does this. Then make the init also use this